pkg/kubectl/cmd/set/set_env.go

```diff
@@ -418,7 +431,7 @@ func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
 		}
 
 		if o.Local || o.DryRun {
-			if err := cmdutil.PrintObject(o.Cmd, patch.Info.AsVersioned(), o.Out); err != nil {
```

Comment:
@deads2k the call to PrintObject here was actually using the humanreadable printer and the no-headers flag.
Before this patch, executing the `set env` command with `--dry-run` would result in the following output:

```bash
$ kubectl set env rc/myrc A=B --dry-run
NAME        AGE
myrc        36m
```

With this patch, it now goes through the success printer:

```bash
$ kubectl set env rc/myrc A=B --dry-run
replicationcontroller/myrc env updated (dry run)
```